app/src/main/java/org/mozilla/fenix/utils/ScreenshotsGuardian.kt

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.utils
+
+import android.view.WindowManager
+import androidx.appcompat.app.AppCompatActivity
+import org.mozilla.fenix.ext.settings
+
+class ScreenshotsGuardian {
+    operator fun invoke(activity: AppCompatActivity) {
```

Comment:
Great choice to extract this responsibility in a separate object that can be freely used from anywhere in the app!
I'm thinking just that `ScreenshotsGuardian.invoke()` is not really telling about what will happen.
Maybe `ScreenshotsGuardian.updateScreenshotsPermission()` would be more indicative of the end result?

An alternative would be to write this method as an extension function to Activity, similar to `Activity.enterToImmersiveMode()`